pkg/jobrunaggregator/jobrunaggregatorlib/util.go

```diff
@@ -112,15 +132,110 @@ func WaitAndGetAllFinishedJobRuns(ctx context.Context,
 			case <-time.After(10 * time.Minute):
 				continue
 			case <-ctx.Done():
-				return finishedJobRuns, unfinishedJobRuns, finishedJobRunNames, unfinishedJobRunNames, ctx.Err()
+				return ctx.Err()
 			}
 		}
 
 		break
 	}
+	return nil
+}
+
+type ClusterJobRunWaiter struct {
+	ProwJobClient      *prowjobclientset.Clientset
+	TimeToStopWaiting  time.Time
+	ProwJobMatcherFunc ProwJobMatcherFunc
+	nextLogTime        time.Time
+}
+
+func (w *ClusterJobRunWaiter) allProwJobFinished(allItems []*prowv1.ProwJob) bool {
+	uncompletedJobMap := map[string]*prowv1.ProwJob{}
+	for _, prowJob := range allItems {
+		if !w.ProwJobMatcherFunc(prowJob) {
+			continue
+		}
+		if prowJob.Status.CompletionTime != nil {
+			continue
+		}
+
+		jobRunID := prowJob.Labels[prowJobJobRunIDLabel]
+		uncompletedJobMap[jobRunID] = prowJob
+	}
+	if len(uncompletedJobMap) == 0 {
+		return true
+	}
+	logrus.Infof("Current time %+v and next lgo time: [%v]", time.Now(), w.nextLogTime)
+	if time.Now().After(w.nextLogTime) {
+		logrus.Infof("The following jobs are not complete: [%v]", strings.Join(sets.StringKeySet(uncompletedJobMap).List(), ", "))
```

Comment:
Please log # of completed jobs as well as not complete just in case there's ever a mismatch. Something like:

"x/y jobs completed, waiting for: [%v]"